crates/resolver/src/tls/dns_over_rustls.rs

```diff
@@ -72,8 +70,6 @@ pub(crate) static CLIENT_CONFIG: Lazy<Result<Arc<ClientConfig>, ProtoError>> = L
     // The port (853) of DOT is for dns dedicated, SNI is unnecessary. (ISP block by the SNI name)
     client_config.enable_sni = false;
 
-    client_config.alpn_protocols.push(ALPN_H2.to_vec());
```

Comment:
hm, I think we might have been doing the same thing in the server, https://github.com/bluejekyll/trust-dns/blob/main/crates/proto/src/rustls/tls_server.rs#L103

I'm not sure why I did that, but I think we'll need to thread that through properly for the DoH settings?

Reply:
> hm, I think we might have been doing the same thing in the server, https://github.com/bluejekyll/trust-dns/blob/main/crates/proto/src/rustls/tls_server.rs#L103

I can look into it, should I do this here or in a different PR?
> I'm not sure why I did that, but I think we'll need to thread that through properly for the DoH settings?

The client already inserts the right ALPN when it finds that none were specified in all protocols so far:
https://github.com/bluejekyll/trust-dns/blob/90edda6b0558282abf852494d7c4f4004e47e1d3/crates/proto/src/https/https_client_stream.rs#L311-L316
https://github.com/bluejekyll/trust-dns/blob/90edda6b0558282abf852494d7c4f4004e47e1d3/crates/proto/src/quic/quic_client_stream.rs#L245-L247

Reply:
So this merely removes the H2 ALPN from the client config. Doesn't this negatively impact the H2 setup? What do DoH (2) standards say on the use of ALPN?

Reply:
> Doesn't this negatively impact the H2 setup?

It has no impact on the H2 setup.
https://github.com/bluejekyll/trust-dns/blob/90edda6b0558282abf852494d7c4f4004e47e1d3/crates/proto/src/https/https_client_stream.rs#L311-L316
This part of the code inserts the required ALPN when no ALPN was found. Removing the initial DoH ALPN was necessary to make other protocols which require different ALPNs work, namely DoQ and DoH3.

> What do DoH (2) standards say on the use of ALPN?

I don't believe DoH has anything to say about ALPN, but HTTP/2 [requires it](https://www.rfc-editor.org/rfc/rfc9113.html#name-http-2-version-identificati).

Reply:
I think that's right, though DoH only works with HTTP/2, so for DoH we should be setting it. Just not for TLS.

Reply:
> [..], so for DoH we should be setting it. Just not for TLS.

Great! That's what we are doing at the moment.
Gonna mark this as resolved then.